plugins/services/src/js/reducers/serviceForm/JSONReducers/__tests__/UnknownVolumes-test.js

```diff
@@ -0,0 +1,137 @@
+const { ADD_ITEM } = require("#SRC/js/constants/TransactionTypes");
+const UnknownVolumes = require("../UnknownVolumes");
+
+describe("UnknownVolumes", function() {
+  describe("#JSONParser", function() {
+    describe("with empty input", function() {
+      it("returns an empty array", function() {
+        expect(UnknownVolumes.JSONParser({})).toEqual([]);
+      });
+    });
+
+    describe("with known volumes", function() {
+      describe("external volume", function() {
+        it("returns an empty array", function() {
+          const externalVolume = {
+            container: {
+              volumes: [
+                {
+                  containerPath: "/mnt/volume",
+                  external: {
+                    name: "someVolume",
+                    provider: "dvdi",
+                    options: {
+                      "dvdi/driver": "rexray"
+                    }
+                  },
+                  mode: "RW"
+                }
+              ]
+            }
+          };
+          expect(UnknownVolumes.JSONParser(externalVolume)).toEqual([]);
+        });
+      });
+
+      describe("persistent volume", function() {
+        it("returns an empty array", function() {
+          const persistentVolume = {
+            container: {
+              volumes: [
+                {
+                  containerPath: "/dev/null",
+                  persistent: { size: 1024 },
+                  mode: "RW"
+                }
+              ]
+            }
+          };
+          expect(UnknownVolumes.JSONParser(persistentVolume)).toEqual([]);
+        });
+      });
+
+      describe("docker volume", function() {
+        it("returns an empty array", function() {
+          const dockerVolume = {
+            container: {
+              volumes: [
+                {
+                  hostPath: "some/path",
+                  containerPath: "/mnt/volume",
+                  mode: "RW"
+                }
+              ]
+            }
+          };
+          expect(UnknownVolumes.JSONParser(dockerVolume)).toEqual([]);
+        });
+      });
+    });
+
+    describe("with unknown volume", function() {
+      it("extracts the volume", function() {
+        const unknownVolumes = {
+          container: {
+            volumes: [
+              {
+                containerPath: "/dev/null"
+              }
+            ]
+          }
+        };
+        expect(UnknownVolumes.JSONParser(unknownVolumes)).toEqual([
+          {
+            type: ADD_ITEM,
+            value: {
+              containerPath: "/dev/null"
+            },
+            path: ["unknownVolumes"]
+          }
+        ]);
```

Comment:
Makes sense but following the link it actually should be:

```javascript
it("extracts the volume", function() {
  // Arrange
  const input = {
    container: {
      volumes: [
        {
          containerPath: "/dev/null"
        }
      ]
    }
  };

  // Act
  const result = UnknownVolumes.JSONParser(input);

  // Assert
  expect(result).toEqual([
    {
      type: ADD_ITEM,
      value: {
        containerPath: "/dev/null"
      },
      path: ["unknownVolumes"]
    }
  ]);
});
```
What do you think?